### ISSUE TITLE
Trino WITHIN GROUP (...) FILTER (...) parsing error

### ISSUE THREAD

**Issue by rileymcdowell** [sqlfluff/sqlfluff#6055]
### Search before asking

- [X] I searched the [issues](https://github.com/sqlfluff/sqlfluff/issues) and found no similar issues.


### What Happened

I wrote some Trino SQL with a `LISTAGG(x, ',') WITHIN GROUP (ORDER BY x) FILTER (x IS NOT NULL)` aggregate function. This generated a parsing error.

I suspected a lack of support for `FILTER`, but testing confirmed this parses fine:

```sql
COUNT(*) FILTER (WHERE o IS NOT NULL) AS count
```

I already knew that `WITHIN GROUP (ORDER BY ...)` was parsable, since I submitted that via a PR a few months ago https://github.com/sqlfluff/sqlfluff/issues/5583. Thank you for accepting that!

So specifically what doesn't parse is the combination of `WITHIN GROUP (...) FILTER (...)`

### Expected Behaviour

I expect `LISTAGG(...) WITHIN GROUP (...) FILTER (...)` to be parsable, and that syntax should be allowable with the default linting configuration for the trino dialect.

### Observed Behaviour

Here's an example SQL file. Aggregate functions require a table, so there's some noise at the top with a bunch of --noqa comments that create an in-memory temp table. That can be ignored.

```sql
WITH t as (                            -- noqa
    SELECT * FROM (                    -- noqa
        VALUES                         -- noqa
            (100, 2, 'a'),             -- noqa
            (100, 1, 'b'),             -- noqa
            (200, NULL, 'c'),          -- noqa
            (200, 2, 'a'),             -- noqa
            (300, NULL, 'b'),          -- noqa
            (300, 1, 'c')              -- noqa
        ) AS t(id, o, value)           -- noqa
)                                      -- noqa

SELECT
    id,
    LISTAGG(value, ',') WITHIN GROUP (ORDER BY o) FILTER (WHERE o IS NOT NULL)
FROM t
GROUP BY id
ORDER BY id;
```

The output from this script should be 

| id | csv_value |
|---|------------|
| 100 | b,a |
| 200 | a |
| 300 | c |

Here's my SQLFluff command:

```bash
sqlfluff lint --dialect trino ./query.sql
```

And here's the output:

```txt
== [query.sql] FAIL                                                                                                                                                                                                         
L:  15 | P:  51 | AL02 | Implicit/explicit aliasing of columns.
                       | [aliasing.column]
L:  15 | P:  51 | CP02 | Unquoted identifiers must be consistently lower case.
                       | [capitalisation.identifiers]
L:  15 | P:  51 | RF04 | Keywords should not be used as identifiers.
                       | [references.keywords]
L:  15 | P:  57 | LT02 | Expected line break and no indent before '('.
                       | [layout.indent]
L:  15 | P:  58 |  PRS | Line 15, Position 58: Found unparsable section: '(WHERE
                       | o IS NOT NULL)'
WARNING: Parsing errors found and dialect is set to 'trino'. Have you configured your dialect correctly?
All Finished 📜 🎉!
```

### How to reproduce

Create a one-line SQL file called query.sql with the following contents:

```sql
WITH t as (                            -- noqa
    SELECT * FROM (                    -- noqa
        VALUES                         -- noqa
            (100, 2, 'a'),             -- noqa
            (100, 1, 'b'),             -- noqa
            (200, NULL, 'c'),          -- noqa
            (200, 2, 'a'),             -- noqa
            (300, NULL, 'b'),          -- noqa
            (300, 1, 'c')              -- noqa
        ) AS t(id, o, value)           -- noqa
)                                      -- noqa

SELECT
    id,
    LISTAGG(value, ',') WITHIN GROUP (ORDER BY o) FILTER (WHERE o IS NOT NULL)
FROM t
GROUP BY id
ORDER BY id;
```

Then run the following command with the latest SQLFluff release:

```bash
sqlfluff lint --dialect trino ./query.sql
```

### Dialect

Trino

### Version

```txt
sqlfluff --version
sqlfluff, version 3.1.0
```

```txt
python --version
python 3.9.16
```

### Configuration

No configuration file was used. This can be triggered with the default configuration.

### Are you willing to work on and submit a PR to address the issue?

- [X] Yes I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://github.com/sqlfluff/sqlfluff/blob/main/CODE_OF_CONDUCT.md)
